src/history/declarativehistorymodel.cpp

```diff
@@ -39,6 +39,20 @@ void DeclarativeHistoryModel::clear()
     emit countChanged();
 }
 
+void DeclarativeHistoryModel::remove(int index)
+{
+    if (index < 0 || index > m_links.count()) {
```

Comment:
Yes it should, blindly copy-pasted a piece of old code. There seems to be a potential bug in data method as well. :)